src/elastio-snap.c

```diff
@@ -1898,7 +2166,22 @@ static int real_fallocate(struct file *f, uint64_t offset, uint64_t length){
 }
 #endif
 
-static int file_allocate(struct file *f, uint64_t offset, uint64_t length){
+static void test_rw(struct snap_device *dev)
+{
+	char *buf = kmalloc(PAGE_SIZE, GFP_KERNEL);
+	int i = 0;
+	for (i = 0; i < 256; i++)
+		memset(buf + i*16, i, 16);
+
+	print_hex_dump(KERN_CONT, "initial buf ", DUMP_PREFIX_OFFSET, 16, 1, buf, PAGE_SIZE, false);
+	file_write_block(dev, buf, 1052672, SECTORS_PER_BLOCK);
+	memset(buf, 0, PAGE_SIZE);
+	file_read_block(dev, buf, 1052672, SECTORS_PER_BLOCK);
+	print_hex_dump(KERN_CONT, "read buf ", DUMP_PREFIX_OFFSET, 16, 1, buf, PAGE_SIZE, false);
+	kfree(buf);
+}
```

Comment:
Again, it seems to be something for debug. Can we use it just in case if module is loaded with `debug=1`?